### PR TITLE
Update FileSystemSyncAccessHandle async methods to sync

### PR DESCRIPTION
**Pull request by dslee414**
Per discussions from https://github.com/whatwg/fs/issues/7 and https://github.com/whatwg/fs/issues/28, this PR updates FileSystemSyncAccessHandle getSize(), truncate(), flush() and close() asynchronous methods to synchronous.

- [x] At least two implementers are interested (and none opposed):
   * Chrome is in the process of shipping
   * Mozilla is interested
   * None opposed
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * Tests are in draft and to be merged upon approval
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1338340
   * Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1797579
   * Safari: https://bugs.webkit.org/show_bug.cgi?id=247071

***
<a href="https://whatpr.org/fs/55.html" title="Last updated on Oct 26, 2022, 5:42 PM UTC (1c3e3a4)">Preview</a> | <a href="https://whatpr.org/fs/55/ad813af...1c3e3a4.html" title="Last updated on Oct 26, 2022, 5:42 PM UTC (1c3e3a4)">Diff</a>

### CODE REVIEW

**Review by a-sully — APPROVED** [review]
LGTM! @annevk please feel free to merge this if it looks good to you. Thanks!

File: index.bs
Comment (annevk) [inline index.bs:1193]:
```suggestion
1. Return [=this=].[=FileSystemSyncAccessHandle/[[file]]=]'s [=file entry/binary data=]'s [=byte sequence/length=].
```

**Review by annevk — APPROVED** [review]
Before merging it's good practice to have all the checkboxes in OP checked. Could someone please file the bugs?

**Comment by a-sully** [comment]
Filed https://bugs.webkit.org/show_bug.cgi?id=247071 and https://bugzilla.mozilla.org/show_bug.cgi?id=1797579

**Pull request merged**
